src/commands/createNewProject/ProjectCreateStep/PowerShellProjectCreateStep.ts

```diff
@@ -54,9 +73,95 @@ export class PowerShellProjectCreateStep extends ScriptProjectCreateStep {
             await fse.writeFile(profileps1Path, profileps1);
         }
 
+        const majorVersion: number | undefined = await this.getLatestAzModuleMajorVersion();
+        if (majorVersion !== undefined) {
+            progress.report({
+                message: `Successfully retrieved ${this.azModuleName} information from PowerShell Gallery"`,
+                increment: 1
+            });
+        } else {
+            progress.report({
+                message: `Failed to retrieve ${this.azModuleName} information from PowerShell Gallery, creating offline template"`,
+                increment: 1
+            });
+        }
+
         const requirementspsd1Path: string = path.join(context.projectPath, requirementspsd1FileName);
         if (await confirmOverwriteFile(requirementspsd1Path)) {
-            await fse.writeFile(requirementspsd1Path, requirementspsd1);
+            if (majorVersion !== undefined) {
+                await fse.writeFile(requirementspsd1Path, requirementspsd1(majorVersion));
+            } else {
+                await fse.writeFile(requirementspsd1Path, requirementspsd1Offine);
+            }
+        }
+    }
+
+    private async getLatestAzModuleMajorVersion(): Promise<number | undefined> {
+        const xmlResult: string | undefined = await this.getPSGalleryAzModuleInfo();
+        if (!xmlResult) {
+            return undefined;
+        }
+
+        const versionResult: string | undefined = await this.parseLatestAzModuleVersion(xmlResult);
+        if (!versionResult) {
+            return undefined;
         }
+
+        try {
+            const [major]: string[] = versionResult.split('.');
+            return parseInt(major);
+        } catch {
+            return undefined;
+        }
+    }
+
+    private async getPSGalleryAzModuleInfo(): Promise<string | undefined> {
+        const request: requestUtils.Request = (
```

Comment:
Could you please add a message that we are connecting to the PowerShell Gallery?